From 36c436144a84a5f5b5c41546cec4dd61095ce698 Mon Sep 17 00:00:00 2001
Message-Id: <36c436144a84a5f5b5c41546cec4dd61095ce698.1415880041.git.chang-joon.lee@intel.com>
In-Reply-To: <4ea77e6cfc48770ef85ec77ae3cdde5d8482417c.1415880041.git.chang-joon.lee@intel.com>
References: <4ea77e6cfc48770ef85ec77ae3cdde5d8482417c.1415880041.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 5 Nov 2014 22:54:19 +0530
Subject: [PATCH 4/5] FOR_UPSTREAM [VPG]: drm/i915: Update the PM interrupts
 before updating the freq

Currently we update the freq before masking the interrupts, which can allow new
interrupts to occur before the frequency has changed. These extra interrupts are
simply wasting cpu cycles. This patch corrects this by masking interrupts prior
to updating the frequency.

Issue: GMINL-3702
Change-Id: Iee460bef43cb7debef573e36b83bd0a3c2315135
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 630091a..29f9979 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3939,6 +3939,8 @@ void valleyview_set_rps(struct drm_device *dev, u8 val)
 	if (IS_CHERRYVIEW(dev))
 		chv_update_rps_cur_delay(dev_priv);
 
+	I915_WRITE(GEN6_PMINTRMSK, gen6_rps_pm_mask(dev_priv, val));
+
 	DRM_DEBUG_DRIVER("GPU freq request from %d MHz (%u) to %d MHz (%u)\n",
 			 vlv_gpu_freq(dev_priv, dev_priv->rps.cur_freq),
 			 dev_priv->rps.cur_freq,
@@ -3947,7 +3949,6 @@ void valleyview_set_rps(struct drm_device *dev, u8 val)
 	if (val != dev_priv->rps.cur_freq)
 		vlv_punit_write(dev_priv, PUNIT_REG_GPU_FREQ_REQ, val);
 
-	I915_WRITE(GEN6_PMINTRMSK, gen6_rps_pm_mask(dev_priv, val));
 	dev_priv->rps.cur_freq = val;
 	trace_intel_gpu_freq_change(vlv_gpu_freq(dev_priv, val));
 }
-- 
1.7.9.5

